components/download_images/src/main.py

```diff
@@ -123,8 +124,9 @@ def transform(
             max_aspect_ratio=max_aspect_ratio,
         )
 
-        # Remove duplicates from laion retrieval
+        # Remove duplicates from laion retrieval (global)
         dataframe = dataframe.drop_duplicates()
```

Comment:
Ideally yes, but I think the issue there is that we are working on a per-partition basis since it's a pandas component. So we are unable to execute `drop duplicates` on the whole dataset afaik

Reply:
Makes sense, but I don't think `drop_duplicates` is very performant until we sort on the index first, which is probably something we should do after retrieving LAION, since we're changing the ids.

Reply:
would that sorting be global then and require repartitioning such that the duplicate ids would all be clustered in one partition? or is it per-partition?

Reply:
Global indeed so they are clustered per partition. I think the `drop_duplicates` method requires a reshuffle step otherwise anyway. Have you generated a task graph of this component? This might be something interesting to log for each component actually.

Reply:
I was trying out the download images component for Datacomp and I had to remove `drop_duplicates` to get it to work

Reply:
why is that? can you elaborate?

Reply:
might be because you have a numpy array in your dataframe and they considered mutable and not hashable. `drop_duplicates()` probably expects the elements to be hashable and by default it's now operating on all the columns, you can specify that you only want to drop an id column by using the `subset` argument as mentioned [here](https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.drop_duplicates.html)

I will also make the changes in this PR